src/peak_event.c

```diff
@@ -19,6 +21,12 @@ peak_event_update(tsd_t *tsd) {
 	uint64_t dalloc = tsd_thread_deallocated_get(tsd);
 	peak_t *peak = tsd_peakp_get(tsd);
 	peak_update(peak, alloc, dalloc);
+
+	activity_callback_thunk_t *thunk = tsd_activity_callback_thunkp_get(
+	    tsd);
+	if (thunk->callback != NULL) {
+		thunk->callback(thunk->uctx, alloc, dalloc);
+	}
```

Comment:
Move these to "peak_alloc_event_handler"? It feels more natural to me there. Or do we explicitly call _event_update somewhere that also requires triggering callback?

Reply:
Do you mean this whole function? The peak_update call and the thunk call sort of logically go together.

I think the structure here might have been copied from the other event handlers? No strong opinions, other than that either both should move or both should stay here.

Reply:
Oh I meant just the new lines. IIRC, the event handler is only called when the event fires, while the event_update()'s job is only to update the threshold for the next event (in some rare cases like reentrancy, event_update may be called w/o triggering the event).

Reply:
Wait I think I was confused.

Reply:
Ah so, the `peak_event_update` is not part of the thread event machinery? I thought that it's part of the thread_event module functions like the event handler.

Reply:
Okay so now the question is, do we want to trigger the callback with the mallctl here: https://github.com/jemalloc/jemalloc/blob/ea32060f9ca5e14077cda7fa2401a1f91f55ad82/src/ctl.c#L2084

Seems not?

Reply:
Ahhh I think it's coming back to me. I think we do want to trigger it there. This is code-sharing between CTL and the thread_event triggered activity updates. Doing the update at CTL time makes it so that calls are always exact if we're at a peak during a thread.peak.read call, which is sometimes handy.

Reply:
Yep makes sense.

Reply:
Oh wait I meant that as a reason why the activity callback ought to be moved (no reason to invoke it at the point of a mallctl call after all). I moved it in a race; want me to move it back?

Reply:
Nope I think we are on the same page: updating the counters on mallctl, but not invoking the callbacks.